### PR TITLE
Remove the v1 API

### PR DESCRIPTION
**Pull request by davidfischer**
- Remove the v1 API
- No longer rely on tastypie

## Open questions

- ~~The embed API is still under `/api/v1/embed/`. This does not remove that. Perhaps that endpoint should be moved or mirrored under `/api/v2`? The embed API does not use tastypie.~~ Update: Moved to `/api/v2/embed/`...

### CODE REVIEW

**Comment by stsewd** [comment]
Shouldn't we at least left the docs page with a warning explaining that v1 was completely removed? Maybe linking to the post or redirect to v2/v3?

**Comment by stsewd** [comment]
> The embed API is still under `/api/v1/embed/`. This does not remove that. Perhaps that endpoint should be moved or mirrored under `/api/v2`? The embed API does not use tastypie.

There are some discussions about that in https://github.com/rtfd/readthedocs.org/issues/5207 and https://github.com/rtfd/readthedocs.org/issues/2667#issuecomment-442248352

**Comment by davidfischer** [comment]
> Shouldn't we at least left the docs page with a warning explaining that v1 was completely removed? Maybe linking to the post or redirect to v2/v3?

I'm not opposed to leaving the v1 doc and putting a large note there that it has been removed. It also should *not* be in the toctree.

**Comment by davidfischer** [comment]
At the same time, I'm not sure the v1 docs are that useful. After this PR, they won't work at all.

File: readthedocs/projects/management/commands/import_project_from_live.py [hunk 1]
```diff
@@ -1,72 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (ericholscher) [inline readthedocs/projects/management/commands/import_project_from_live.py]: I still use this helper. I think it should be pretty simple to adopt it to v2.

Comment (stsewd) [reply readthedocs/projects/management/commands/import_project_from_live.py]: ha, I use it too :(

**Comment by davidfischer** [comment]
> I do wonder about rolling this out slowly, if we can. Perhaps doing a brownout of these API's at the Nginx level for a day, then a week; similar to what GH did for webhooks. That would give anyone still depending on it time to migrate before we totally kill it.

I think this is a good idea.

**Comment by davidfischer** [comment]
> I think it's worth keeping the `v1.rst` around with a note that it's been removed. We don't need the old content, but it shouldn't just "disappear".

I re-added the page although it's blank except for noting that the API has been removed. I think that's good in case people have links to the API or whatnot but I think the content of the page ceases to have value once the API is turned off.

**Comment by davidfischer** [comment]
Also, the embed API has now been mirrored to https://readthedocs.org/api/v2/embed/.

File: readthedocs/projects/management/commands/import_project_from_live.py (hunk 1 above)
Comment (humitos) [reply readthedocs/projects/management/commands/import_project_from_live.py]: Yes. This command is useful when debugging production projects locally.

I'd like to find a way to keep it.

Comment (davidfischer) [reply readthedocs/projects/management/commands/import_project_from_live.py]: I'll update this with an adapted version using the v2 API.

Comment (stsewd) [reply readthedocs/projects/management/commands/import_project_from_live.py]: We can do it in another PR, tho :)

Comment (davidfischer) [reply readthedocs/projects/management/commands/import_project_from_live.py]: Nah, I think it isn't too onerous... Also, that way the single PR has the full set of changes.

Comment (davidfischer) [reply readthedocs/projects/management/commands/import_project_from_live.py]: I re-added this although there are some changes. The biggest one is that in order to get all the fields from an API response, the request needs to be authenticated and the user must be staff ([See this](https://github.com/rtfd/readthedocs.org/blob/3218c808e87206983c4eb3ce4faceddbcc0f7aa3/readthedocs/restapi/views/model_views.py#L82-L95)). One key field that is only returned with the `ProjectAdminSerializer` is the `requirements_file`.

By default, the management command will be unauthenticated. However, if you have `READ_THE_DOCS_USERNAME` and `READ_THE_DOCS_PASSWORD` in your environment variables when you execute the management command, they'll be used.

File: readthedocs/projects/management/commands/import_project_from_live.py [hunk 2]
```diff
@@ -32,38 +33,62 @@ def add_arguments(self, parser):
...
+                'features',
+                'environment_variables',
             )
 
             for attribute in project_data:
                 if attribute not in exclude_attributes:
                     setattr(project, attribute, project_data[attribute])
+                    self.stdout.write(' - Setting {key} to {val}'.format(
+                        key=attribute,
+                        val=project_data[attribute]),
```
Comment (humitos) [inline readthedocs/projects/management/commands/import_project_from_live.py]: This one is a nice improvement!

File: readthedocs/projects/management/commands/import_project_from_live.py [hunk 3]
```diff
@@ -32,38 +33,62 @@ def add_arguments(self, parser):
         parser.add_argument('project_slug', nargs='+', type=str)
 
     def handle(self, *args, **options):
-        api = slumber.API(base_url='http://readthedocs.org/api/v1/')
+        auth = None
         user1 = User.objects.filter(pk__gt=0).order_by('pk').first()
 
+        if 'READ_THE_DOCS_USERNAME' in os.environ and 'READ_THE_DOCS_PASSWORD' in os.environ:
```
Comment (humitos) [inline readthedocs/projects/management/commands/import_project_from_live.py]: I have a _personal_ preference for `READTHEDOCS_USERNAME` (to follow the same pattern from https://docs.readthedocs.io/en/latest/builds.html#build-environment).

Comment (davidfischer) [reply readthedocs/projects/management/commands/import_project_from_live.py]: I think that's pretty reasonable to stay consistent.

File: docs/api/index.rst [hunk 4]
```diff
@@ -5,12 +5,6 @@ This section of the documentation details the public API
 usable to get details of projects, builds, versions and other details
 from Read the Docs.
 
-.. warning::
-
-    Originally, the Read the Docs API allowed connections over insecure HTTP.
-    Starting in January 2019, requests over HTTP
-    will be automatically redirected to HTTPS
-    and non-GET/HEAD requests over insecure HTTP will fail.
```
Comment (davidfischer) [inline docs/api/index.rst]: I removed this while I was here, but technically this change is not due to API v1 but rather due to https://github.com/rtfd/readthedocs-ops/pull/461.

**Comment by stsewd** [comment]
Looks like we can remove more dependencies here

https://github.com/rtfd/readthedocs.org/pull/5389/files#r262275644

**Comment by stsewd** [comment]
And guessing that we can remove some dependencies from the installation guide 

https://github.com/rtfd/readthedocs.org/blob/master/docs/install.rst#L41-L42

**Comment by davidfischer** [comment]
We've been returning failures for API v1 for over a week and I haven't heard anything. I think this is ok to go. I'll see if I can remove those deps.

**Comment by davidfischer** [comment]
> I'll see if I can remove those deps.

lxml is required by pyquery but I'll update the comments and move it to a more appropriate spot in the requirements file.

**Comment by davidfischer** [comment]
I should also add that while defusedxml is not strictly required, I think that we should install defusedxml as long as lxml is used. DefusedXML is a library that attempts to mitigate or remove XML parsing vulnerabilities and in many cases it monkeypatches other libraries.

**Pull request merged**